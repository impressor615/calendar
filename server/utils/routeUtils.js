const errors = require('../errors');

module.exports = {
  sendError: ({
    res,
    language,
    error = errors.route_invalid_data,
    status = 400,
  }) => {
    const message = error[language] || error.en;
    return res.status(status).json({
      type: 'error',
      message,
    });
  },
};
