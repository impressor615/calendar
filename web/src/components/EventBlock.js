import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Button from 'components/Button';


const EventBlock = ({
  children,
  onToggle,
  onDragStart,
  fluid,
  ...rest
}) => (
  <Button
    {...rest}
    draggable
    onClick={onToggle}
    onDragStart={onDragStart}
    className={classnames(
      'event-block',
      rest.className,
      { fluid },
    )}
  >
    { children }
  </Button>
);

EventBlock.defaultProps = {
  fluid: false,
};

EventBlock.propTypes = {
  children: PropTypes.node.isRequired,
  onToggle: PropTypes.func.isRequired,
  onDragStart: PropTypes.func.isRequired,
  fluid: PropTypes.bool,
}

export default EventBlock;