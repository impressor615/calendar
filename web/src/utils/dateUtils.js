import moment from 'moment';

export const monthRange = (date) => {
  const startDate = moment(date).startOf('month').startOf('week');
  const endDate = moment(date).endOf('month').endOf('week');
  const diff = endDate.diff(startDate, 'days') + 1;
  return {
    start_date: startDate,
    end_date: endDate,
    items: Array.from({ length: diff }).map((_, index) => {
      const targetDate = moment(startDate).add(index, 'days');
      return {
        year: targetDate.year(),
        month: (targetDate.month() + 1),
        date: targetDate.date(),
        moment: targetDate,
      };
    }),
  }
};

export const weekRange = (date) => {
  const startDate = moment(date).startOf('week');
  const endDate = moment(date).endOf('week');
  const diff = endDate.diff(startDate, 'days') + 1;
  return {
    start_date: startDate,
    end_date: endDate,
    items: Array.from({ length: diff }).map((_, index) => {
      const targetDate = moment(startDate).add(index, 'days');
      return {
        year: targetDate.year(),
        month: (targetDate.month() + 1),
        date: targetDate.date(),
        moment: targetDate,
      };
    }),
  };
};

export const getHourKey = (hour) => {
  let hourKey = hour.toString();
  if (hour <= 9) {
    hourKey = `0${hourKey}`;
  }
  return hourKey;
};
