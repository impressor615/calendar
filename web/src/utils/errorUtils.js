import CONSTANTS from '../constants';

const { ERRORS } = CONSTANTS;
export const getErrorMsg = (error) => {
  const { response } = error.request;
  const errorObj = JSON.parse(response);
  const msg = ERRORS[errorObj.code].ko || ERRORS['401'].ko;
  return msg;
}