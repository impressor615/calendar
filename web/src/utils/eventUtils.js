export const eventBind = (onEvent, props) => event => onEvent(event, props);