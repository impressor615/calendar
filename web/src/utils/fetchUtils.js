import axios from 'axios';


export const authAxios = axios.create({ headers: { 'x-access-token': process.env.REACT_APP_SECRET } });

export const fetchEvents = (startDate, endDate) => {
  const start_date = startDate.toISOString();
  const end_date = endDate.toISOString();
  return authAxios.get('/api/calendar', { params: { start_date, end_date }})
    .then(res => res.data)
};

export const updateServerEvent = (_id, putData, events) => (
  authAxios.put(`/api/calendar/${_id}`, putData)
    .then(() => {
      const newEvents = [...events].filter(item => item._id !== _id);
      putData._id = _id
      newEvents.push(putData);
      return newEvents;
    })
);

