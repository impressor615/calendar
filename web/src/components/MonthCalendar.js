import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import EventBlock from 'components/EventBlock';
import { getHourKey } from 'utils/dateUtils';
import { eventBind } from 'utils/eventUtils';
import CONSTANTS from '../constants';


const { DAY } = CONSTANTS;
export const CalendarDay = ({
  day,
  dateObj,
  children,
  currentDate,
  onToggle,
  onDrop,
}) => {
  const isYearShowing = currentDate.year() !== dateObj.year;
  const isMonthShowing = (currentDate.month() + 1) !== dateObj.month;
  return (
    <div
      className="cal-block"
      onClick={eventBind(onToggle, { type: 'month', dateObj })}
      onDrop={eventBind(onDrop, { type: 'month', dateObj })}
      onDragOver={eventBind((e) => e.preventDefault())}
    >
      { day ? <div className="day">{day}</div> : null}
      <div>
        { isYearShowing ? `${dateObj.year}년 ${dateObj.month}월 ${dateObj.date}일` : null }
        { !isYearShowing && isMonthShowing ? `${dateObj.month}월 ${dateObj.date}일` : null }
        { !isYearShowing && !isMonthShowing ? `${dateObj.date}` : null }
      </div>
      {children}
    </div>
  );
}

CalendarDay.defaultProps = {
  day: '',
  children: '',
};

CalendarDay.propTypes = {
  day: PropTypes.string,
  children: PropTypes.node,
  dateObj: PropTypes.object.isRequired,
  currentDate: PropTypes.object.isRequired,
  onToggle: PropTypes.func.isRequired,
  onDrop: PropTypes.func.isRequired,
};

const MonthCalendar = ({
  range,
  onToggle,
  onDragStart,
  onDrop,
  currentDate,
  data,
}) => (
  <section className="month-calendar">
    {
      range.map((dateObj, index) => {
        const blockDate = dateObj.moment.format('YYYY-MM-DD');
        const events = data.filter((item) => {
          const eventDate = moment(item.start_date).format('YYYY-MM-DD');
          return eventDate === blockDate;
        }).sort((current, next) => new Date(current.start_date) - new Date(next.start_date));
        return (
          <CalendarDay
            key={index.toString()}
            currentDate={currentDate}
            dateObj={dateObj}
            day={DAY[index]}
            onToggle={onToggle}
            onDrop={onDrop}
          >
           {
              events.length
                ? (
                  <Fragment>
                    {
                      events.map((event, index) => (
                        <EventBlock
                          key={index.toString()}
                          className="primary"
                          onDragStart={eventBind(onDragStart, { event })}
                          onToggle={eventBind(onToggle, { type: 'event', dateObj: event })}
                        >
                          &#9675; {`${getHourKey(moment(event.start_date).hours())}:00 ${event.title}`}
                        </EventBlock>
                      ))
                    }
                  </Fragment>
                ) : null
            }
          </CalendarDay>
        );
      })
    }
  </section>
);

MonthCalendar.propTypes = {
  range: PropTypes.arrayOf(PropTypes.object).isRequired,
  onToggle: PropTypes.func.isRequired,
  onDragStart: PropTypes.func.isRequired,
  onDrop: PropTypes.func.isRequired,
  currentDate: PropTypes.object.isRequired,
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default MonthCalendar;