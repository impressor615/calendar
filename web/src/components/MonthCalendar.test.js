import React from 'react';
import moment from 'moment';

import MonthCalendar, { CalendarDay } from 'components/MonthCalendar';
import EventBlock from 'components/EventBlock';
import { monthRange } from 'utils/dateUtils';


describe("<MonthCalendar />", () => {
  const onToggle = jest.fn();
  const onDrop = jest.fn();
  const onDragStart = jest.fn();
  const currentDate = moment('2018-10-14');
  const { range } = monthRange(currentDate);
  const props = {
    range,
    currentDate,
    onToggle,
    onDrop,
    onDragStart,
    data: [
      {
        title: 'title',
        start_date: moment(currentDate).toISOString(),
        end_date: moment(currentDate).add(1, 'hours').toISOString(),
      },
    ]
  };

  it('should be rendered properly', () => {
    const wrapper = shallow(<MonthCalendar {...props} />);
    expect(wrapper.find('.month-calendar')).toHaveLength(1);
    expect(wrapper.find(CalendarDay)).toHaveLength(range.length);
    expect(wrapper.find(EventBlock)).toHaveLength(1);
  });

  it('methods should be called properly', () => {
    const wrapper = shallow(<MonthCalendar {...props} />);
    wrapper.find(CalendarDay).first().simulate('drop');
    wrapper.find(EventBlock).simulate('dragstart');
    expect(onDrop).toHaveBeenCalledTimes(1);
    expect(onDragStart).toHaveBeenCalledTimes(1);
  });

  it('should be matched with the snapshot', () => {
    const wrapper = shallow(<MonthCalendar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
