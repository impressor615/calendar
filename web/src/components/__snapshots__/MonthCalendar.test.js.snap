// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonthCalendar /> should be matched with the snapshot 1`] = `
<section
  className="month-calendar"
>
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 30,
        "moment": "2018-09-29T15:00:00.000Z",
        "month": 9,
        "year": 2018,
      }
    }
    day="일"
    key="0"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 1,
        "moment": "2018-09-30T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day="월"
    key="1"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 2,
        "moment": "2018-10-01T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day="화"
    key="2"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 3,
        "moment": "2018-10-02T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day="수"
    key="3"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 4,
        "moment": "2018-10-03T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day="목"
    key="4"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 5,
        "moment": "2018-10-04T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day="금"
    key="5"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 6,
        "moment": "2018-10-05T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day="토"
    key="6"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 7,
        "moment": "2018-10-06T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="7"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 8,
        "moment": "2018-10-07T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="8"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 9,
        "moment": "2018-10-08T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="9"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 10,
        "moment": "2018-10-09T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="10"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 11,
        "moment": "2018-10-10T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="11"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 12,
        "moment": "2018-10-11T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="12"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 13,
        "moment": "2018-10-12T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="13"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 14,
        "moment": "2018-10-13T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="14"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  >
    <EventBlock
      className="primary"
      fluid={false}
      key="0"
      onDragStart={[Function]}
      onToggle={[Function]}
    >
      ○ 
      00:00 title
    </EventBlock>
  </CalendarDay>
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 15,
        "moment": "2018-10-14T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="15"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 16,
        "moment": "2018-10-15T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="16"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 17,
        "moment": "2018-10-16T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="17"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 18,
        "moment": "2018-10-17T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="18"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 19,
        "moment": "2018-10-18T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="19"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 20,
        "moment": "2018-10-19T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="20"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 21,
        "moment": "2018-10-20T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="21"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 22,
        "moment": "2018-10-21T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="22"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 23,
        "moment": "2018-10-22T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="23"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 24,
        "moment": "2018-10-23T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="24"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 25,
        "moment": "2018-10-24T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="25"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 26,
        "moment": "2018-10-25T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="26"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 27,
        "moment": "2018-10-26T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="27"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 28,
        "moment": "2018-10-27T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="28"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 29,
        "moment": "2018-10-28T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="29"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 30,
        "moment": "2018-10-29T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="30"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 31,
        "moment": "2018-10-30T15:00:00.000Z",
        "month": 10,
        "year": 2018,
      }
    }
    day=""
    key="31"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 1,
        "moment": "2018-10-31T15:00:00.000Z",
        "month": 11,
        "year": 2018,
      }
    }
    day=""
    key="32"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 2,
        "moment": "2018-11-01T15:00:00.000Z",
        "month": 11,
        "year": 2018,
      }
    }
    day=""
    key="33"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
  <CalendarDay
    currentDate={"2018-10-13T15:00:00.000Z"}
    dateObj={
      Object {
        "date": 3,
        "moment": "2018-11-02T15:00:00.000Z",
        "month": 11,
        "year": 2018,
      }
    }
    day=""
    key="34"
    onDrop={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onToggle={[MockFunction]}
  />
</section>
`;
